### ISSUE TITLE
[Bug]: With MongoDB as a data source, NaN comes in as {"$numberDouble":"NaN"}

### ISSUE THREAD

**Issue by markebowles** [appsmithorg/appsmith#29636]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When using MongoDB as a data source, a numeric field in the database with a value NaN is returned to the AppSmith environment as an object, {"$numberDouble":"NaN"}.  This forces the application to code a special case. Doing so is a workaround, but there's no value-added by not converting this structure back to Math.NaN before the application sees it. 

One consequence of this problem is that rounding the value for display in a table is inconveniently complex. Defining the column value as {"$numberDouble":"NaN"}.toFixed(3), for example, fails whenever the NaN value appears, and the entire column goes blank.

### Steps To Reproduce

Create a small collection in MongoDb, and add a document with a value of NaN.  Use AppSmith to auto-create a table on it.

### Public Sample App

none available

### Environment

Production

### Version

Cloud 2023-12-14

**Comment by Nikhil-Nandagopal** [comment]
@markebowles Is the proposal to change NaN to Null? Won't that suffer from the same issue?

**Comment by markebowles** [comment]
No, the proposal is to replace the JSON structure {"$numberDouble":"NaN"} with the Double value, NaN.  I realize this is obscure, but in my app NaN is an expected value in some table columns.  Since the data representation facility in AppSmith is JavaScript, not JSON, I would have expected that by the time the value gets into my app, it's back to being a floating-point number.

I should tell you I'm an utter noobie with AppSmith, but an experienced JS developer, so perhaps I don't understand the system's conventions.

I changed the Computed Value for the column in question to: {{currentRow.ranges.COP.max == Math.NaN}} and the NaN values showed up as false (as did all the other numbers). This feels like inappropriate layer-mixing to me.

This is more of a problem for me as an app developer than it should be, because if the Computed Value is {{currentRow.ranges.COP.max.toFixed(3)}}, then a value of NaN on any page of a table causes he entire column to be blank.

Some charting packages render NaN by not drawing the data point, providing apps with a mechanism to display the notion that "nothing happened at this point in time". Analogous behavior in AppSmith's table would be to display a blank value in specific rows where the Computed Value is undefined.
